src/planwise/component/scenarios.clj

```diff
@@ -38,6 +40,13 @@
     (if (zero? providers) ""
         (format "Create %d %s. Increase overall capacity in %d." providers u capacity))))
 
+(defn- map->csv
+  [file-name coll]
+  {:coll coll}
+  (let [fields (mapv name (keys (first coll)))
+        data   (cons fields (mapv vals coll))]
```

Comment:
While it is not likely to happen if the maps of the collection are created by the same code, there are no guarantees that the order of the keys of the first map will match the order in which `vals` returns the values. Ie. the order for `keys` and `vals` *for the same map* is guaranteed to be the same, but the data is obtained from multiple maps.

Eg.

```clojure
(def a {:id 1 :name "foo"})
(def b {:name "bar" :id 2})
(keys a) ;;=> (:id :name)
(vals a) ;;=> (1 "foo")
(vals b) ;;=> ("bar" 2)
```

Reply:
This is still relevant after the changes in 2cfb7e2